bundle/bin/rke2-uninstall.sh

```diff
@@ -121,7 +121,7 @@ uninstall_remove_files()
     rm -rf /etc/cni
     rm -rf /opt/cni/bin
     rm --one-file-system -rf /var/lib/kubelet || true
-    rm -rf "${RKE2_DATA_DIR}" || error "Failed to remove /var/lib/rancher/rke2"
+    rm -rf -- "${RKE2_DATA_DIR}" || error "Failed to remove ${RKE2_DATA_DIR}"
```

Comment:
> RKE2_DATA_DIR is only defaulted in the script and could be wrongly defined by a user outside of it.

I don't see how adding `--` to terminate flag parsing makes this any safer. If the user sets `RKE2_DATA_DIR=/` it will still happily `rm -rf /`. The only thing this does is prevent `rm` from confusing the data-dir arg with additional flags, if for some reason the path starts with a `-` and I'm not sure that would even work to begin with.

What actual problem does this solve?